tests/EStimLibrary.UnitTests/Core/HardwareInterfaces/NeuralInterfaceManagerTests.cs

```diff
@@ -0,0 +1,207 @@
+using EStimLibrary.Core;
+using EStimLibrary.Core.HardwareInterfaces;
+using EStimLibrary.Extensions.HardwareInterfaces;
+
+
+namespace EStimLibrary.UnitTests.Core;
+
+
+public class NeuralInterfaceManagerTests
+{
+    /// <summary>
+    /// An error is thrown when the Type is invalid, i.e. types are not concrete derived class of NeuralInterfaceHardware.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ReusableIdPool), new object[]{0, 0, new int[0]})]
+    [InlineData(typeof(int), new object[]{})]
+    [InlineData(typeof(NeuralInterfaceHardware), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldThowError_WhenInvalidType(Type interfaceType, object[] interfaceSpecificParams) 
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act & Assert
+        Assert.Throws<ArgumentException>(() => NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _));
+
+    }
+
+    /// <summary>
+    /// Given a valid Type, the manager assigns the next available ID to the new neural interface
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldInit_WhenValidType(Type interfaceType, object[] interfaceSpecificParams) 
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var result = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out int globalInterfaceId);
+        
+        // Assert
+        Assert.NotNull(result);
```

Comment:
If there isn't an available Id for the interface, the method will return an empty set rather than null. In this case though, to result should specifically be the correct list of contact IDs assigned to that neural interface. You can add another parameter (or more!) to your test methods and InlineData values to take in expected values